#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BrushedAluminium.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/geometries/protos/Rectangle.proto"

WorldInfo {
  info [
    "Example use of a color Camera device."
  ]
  title "Camera"
}
Viewpoint {
  orientation -0.22708866491359825 -0.02917256321061952 0.9734370548851538 3.336470746580091
  position 1.962083368434483 0.31479897734060824 0.9670252609908204
}
TexturedBackground {
  skybox FALSE
}
TexturedBackgroundLight {
}
RectangleArena {
  floorAppearance Parquetry {
    type "light strip"
  }
  wallAppearance BrushedAluminium {
    colorOverride 0.190127 0.193362 0.18851
  }
}
PointLight {
  attenuation 0 0 1
  intensity 0.6
  location 0 0 0.6
}
DEF BLUE_BOX Solid {
  translation -0.452684 0.44453 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 0.519341125
        metalness 0
      }
      geometry DEF BOX1 Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "blue box"
  boundingObject Shape {
    appearance PBRAppearance {
      metalness 0
    }
    geometry USE BOX1
  }
}
Robot {
  translation 0.108032 -0.143689 8.88178e-16
  rotation 0 0 1 1.2618
  children [
    TouchSensor {
      translation 0.045 0 0.01
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0
              }
              geometry Box {
                size 0.01 0.09 0.01
              }
            }
          ]
        }
      ]
      name "bumper"
      boundingObject Rectangle {
        size 0.09 0.01
      }
    }
    Pose {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.044999999999999055 0.025
        rotation 0.13448695438559993 -0.13448695438560018 -0.9817466670176024 1.5892169999999615
        children [
          DEF WHEEL Pose {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.044999999999999415 0.025
        rotation -0.06235700094427267 0.062357000944275334 -0.996104015084003 1.574699999999973
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation 0.042 0.02 0.063
      rotation 0 0 -1 -0.499205
      children [
        DEF SENSOR Pose {
          rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation 0 0 1 -0.500795
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DEF SMILE Pose {
      translation 0.036 0 0.025
      rotation 0.25056300542381216 -0.9351130202419243 -0.25056300542381216 1.637834
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
    Camera {
      translation 0.04 0 0.0915
      children [
        Pose {
          translation -0.015 0 0
          rotation -0.5771968951617173 -0.577656895078166 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.03
                radius 0.01
              }
            }
          ]
        }
      ]
      fieldOfView 1.047
      width 80
      height 60
      antiAliasing TRUE
      noise 0.04
    }
  ]
  name "MyBot"
  boundingObject Pose {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "robot_controller"
}
Solid {
  translation 0.187695 0.253608 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Cylinder {
        height 0.1
        radius 0.025
      }
    }
  ]
  name "solid(1)"
  boundingObject Cylinder {
    height 0.1
    radius 0.025
  }
}
Solid {
  translation -0.134023 -0.399497 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        metalness 0
      }
      geometry Cylinder {
        height 0.1
        radius 0.025
      }
    }
  ]
  name "solid(4)"
  boundingObject Cylinder {
    height 0.1
    radius 0.025
  }
}
Solid {
  translation -0.236705 0.137399 0.05
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Cylinder {
        height 0.1
        radius 0.025
      }
    }
  ]
  name "solid(3)"
  boundingObject Cylinder {
    height 0.1
    radius 0.025
  }
}
